ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java

```diff
@@ -192,217 +178,43 @@ void onAdFail(String placement) {
-      if (mediationBannerListener != null) {
-        mediationBannerListener.onAdFailedToLoad(
-            VungleInterstitialAdapter.this, AdRequest.ERROR_CODE_INVALID_REQUEST);
-      }
-      return;
-    }
-
-    mVungleManager = VungleManager.getInstance();
-
-    String placementForPlay = mVungleManager.findPlacement(mediationExtras, serverParameters);
-    Log.d(
-        TAG,
-        "requestBannerAd for Placement: "
-            + placementForPlay
-            + " ###  Adapter instance: "
-            + this.hashCode());
-
-    if (TextUtils.isEmpty(placementForPlay)) {
-      String message = "Failed to load ad from Vungle: Missing or Invalid Placement ID.";
-      Log.w(TAG, message);
-      mMediationBannerListener.onAdFailedToLoad(
-          VungleInterstitialAdapter.this, AdRequest.ERROR_CODE_INVALID_REQUEST);
-      return;
-    }
-
-    AdConfig adConfig = VungleExtrasBuilder.adConfigWithNetworkExtras(mediationExtras, true);
-    if (!hasBannerSizeAd(context, adSize, adConfig)) {
-      String message = "Failed to load ad from Vungle: Invalid banner size.";
-      Log.w(TAG, message);
-      mMediationBannerListener.onAdFailedToLoad(
-          VungleInterstitialAdapter.this, AdRequest.ERROR_CODE_INVALID_REQUEST);
-      return;
-    }
-
-    // Create the adLayout wrapper with the requested ad size, as Vungle's ad uses MATCH_PARENT for
-    // its dimensions.
-    adLayout =
-        new RelativeLayout(context) {
-          @Override
-          protected void onAttachedToWindow() {
-            super.onAttachedToWindow();
-            if (mBannerRequest != null) {
-              mBannerRequest.attach();
-            }
-          }
-
-          @Override
-          protected void onDetachedFromWindow() {
-            super.onDetachedFromWindow();
-            if (mBannerRequest != null) {
-              mBannerRequest.detach();
-            }
-          }
-        };
-    int adLayoutHeight = adSize.getHeightInPixels(context);
-    // If the height is 0 (e.g. for inline adaptive banner requests), use the closest supported size
-    // as the height of the adLayout wrapper.
-    if (adLayoutHeight <= 0) {
-      float density = context.getResources().getDisplayMetrics().density;
-      adLayoutHeight = Math.round(adConfig.getAdSize().getHeight() * density);
-    }
-    RelativeLayout.LayoutParams adViewLayoutParams =
-        new RelativeLayout.LayoutParams(adSize.getWidthInPixels(context), adLayoutHeight);
-    adLayout.setLayoutParams(adViewLayoutParams);
-
-    mBannerRequest =
-        mVungleManager.getBannerRequest(placementForPlay, config.getRequestUniqueId(), adConfig);
-    if (mBannerRequest == null) {
-      // Adapter does not support multiple Banner instances playing for same placement except for
-      // Refresh
-      mMediationBannerListener.onAdFailedToLoad(
-          VungleInterstitialAdapter.this, AdRequest.ERROR_CODE_INVALID_REQUEST);
-      return;
-    }
-
-    mBannerRequest.setAdLayout(adLayout);
```

Comment:
this seems to breaks refresh case, VungleBannerAdapter can be finished by the old instance of VungleInterstitialAdapter